From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Tue, 8 Nov 2022 15:00:41 +0300
Subject: Do not configure in etc subdirectory

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 78f5be6..a59c04b 100644
--- a/Makefile
+++ b/Makefile
@@ -79,7 +79,7 @@ dist:
 # {{{ User level target--  config
 
 config:
-	@cd etc && $(MAKE) config  
+	#@cd etc && $(MAKE) config  
 	@cd lisp && $(MAKE) config
 	@cd lisp/g-client  && $(MAKE) config
 	@echo "Configured emacspeak in directory $(SRC)."
