From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Sat, 12 Nov 2022 11:15:18 +0300
Subject: Correct cleaning commands

---
 info/Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/info/Makefile b/info/Makefile
index 595cde0..279f42c 100644
--- a/info/Makefile
+++ b/info/Makefile
@@ -34,7 +34,7 @@ docs.texi: ../lisp/*.el
 
 emacspeak.pdf: emacspeak.texi docs.texi
 	makeinfo --pdf emacspeak.texi
-	\rm emacspeak.aux emacspeak.cp emacspeak.cps emacspeak.fn \
+	rm -f emacspeak.aux emacspeak.cp emacspeak.cps emacspeak.fn \
 emacspeak.fns emacspeak.ky emacspeak.kys emacspeak.log \
 emacspeak.toc  emacspeak.vr emacspeak.vrs
 
@@ -56,7 +56,8 @@ pub:  html/index.html introducing-emacspeak.html
 
 
 clean: 
-	\rm *.info *.pdf || echo "all clean"
+	rm -f *.info* *.html *.ps *.dvi *.pdf *.txt
+	rm -rf html introducing-emacspeak
 
 spell:
 	@mv .spell.words .spell.tex
