From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Tue, 8 Nov 2022 15:35:11 +0300
Subject: Auditory icon method tags in customization

---
 lisp/emacspeak-sounds.el | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/lisp/emacspeak-sounds.el b/lisp/emacspeak-sounds.el
index 3dd4047..54bc4b5 100644
--- a/lisp/emacspeak-sounds.el
+++ b/lisp/emacspeak-sounds.el
@@ -300,11 +300,11 @@ Native : Use Emacs' builtin sound support.
 Use Serve when working with remote speech servers."
   :group 'emacspeak-sounds
   :type '(choice
-          (const emacspeak-play-auditory-icon)
-          (const emacspeak-serve-auditory-icon)
-          (const emacspeak-play-auditory-icon)
-          (const emacspeak-soxplay-auditory-icon)
-          (const emacspeak-queue-auditory-icon)))
+          (const :tag "Native" emacspeak-native-auditory-icon)
+          (const :tag "Serve" emacspeak-serve-auditory-icon)
+          (const :tag "play" emacspeak-play-auditory-icon)
+          (const :tag "soxplay" emacspeak-soxplay-auditory-icon)
+          (const :tag "Queue" emacspeak-queue-auditory-icon)))
 
 ;;;###autoload
 (defun emacspeak-auditory-icon (icon)
