From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Tue, 8 Nov 2022 15:37:05 +0300
Subject: Auditory icon players table fix

---
 lisp/emacspeak-sounds.el | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lisp/emacspeak-sounds.el b/lisp/emacspeak-sounds.el
index 54bc4b5..093556c 100644
--- a/lisp/emacspeak-sounds.el
+++ b/lisp/emacspeak-sounds.el
@@ -341,8 +341,10 @@ Optional interactive PREFIX arg toggles global value."
 
 (defvar emacspeak-sounds-auditory-icon-players
   '("emacspeak-serve-auditory-icon"
+    "emacspeak-queue-auditory-icon"
+    "emacspeak-native-auditory-icon"
     "emacspeak-play-auditory-icon"
-    "emacspeak-play-auditory-icon")
+    "emacspeak-soxplay-auditory-icon")
   "Table of auditory icon players used  when selecting a player.")
 
 (defun emacspeak-select-auditory-icon-player ()
