From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Tue, 8 Nov 2022 19:21:52 +0300
Subject: Unspeakable rule customization facility

---
 lisp/emacspeak-speak.el | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/lisp/emacspeak-speak.el b/lisp/emacspeak-speak.el
index 89f4f6c..ebfc3b6 100644
--- a/lisp/emacspeak-speak.el
+++ b/lisp/emacspeak-speak.el
@@ -862,13 +862,15 @@ Argument START  and END specify region to speak."
 (put 'emacspeak-decoration-rule 'variable-interactive
      "sEnterregular expression to match lines that are decorative ASCII: ")
 
-(defvar emacspeak-unspeakable-rule
-  "^[^0-9a-zA-Z]+$"
+(defcustom emacspeak-unspeakable-rule
+  "^\\W+$"
   "Pattern to match lines of special chars.
 This is a regular expression that matches lines containing only
 non-alphanumeric characters.  emacspeak will generate a tone
 instead of speaking such lines when punctuation mode is set
-to some.")
+to some."
+  :group 'emacspeak-speak
+  :type 'regexp)
 
 (put 'emacspeak-unspeakable-rule 'variable-interactive
      "sEnterregular expression to match unspeakable lines: ")
