From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Tue, 8 Nov 2022 19:44:26 +0300
Subject: Don't speak messages by default

---
 lisp/emacspeak-speak.el | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lisp/emacspeak-speak.el b/lisp/emacspeak-speak.el
index f8bad06..60135a3 100644
--- a/lisp/emacspeak-speak.el
+++ b/lisp/emacspeak-speak.el
@@ -181,7 +181,7 @@ current local  value to the result.")
 ;;}}}
 ;;{{{ Shell Command Helper:
 
-(defcustom emacspeak-speak-messages t
+(defcustom emacspeak-speak-messages nil
   "Option indicating if messages are spoken.  If nil,
 emacspeak will not speak messages as they are echoed to the
 message area.  You can use command
