From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Wed, 9 Nov 2022 13:23:12 +0300
Subject: General close operation feedback

---
 lisp/emacspeak-advice.el | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/lisp/emacspeak-advice.el b/lisp/emacspeak-advice.el
index 05c6585..1461c02 100644
--- a/lisp/emacspeak-advice.el
+++ b/lisp/emacspeak-advice.el
@@ -291,7 +291,7 @@ When on a close delimiter, speaking matching open delimiter after a small delay.
 
 (cl-loop
  for f in
- '(next-buffer previous-buffer bury-buffer)
+ '(next-buffer previous-buffer)
  do
  (eval
   `(defadvice ,f (after emacspeak pre act comp)
@@ -1729,7 +1729,11 @@ Produce an auditory icon if possible."
     (emacspeak-speak-mode-line)))
 (cl-loop
  for f in
- '(kill-buffer quit-window)
+ '(kill-buffer
+   kill-this-buffer
+   bury-buffer
+   delete-frame
+   quit-window)
  do
  (eval
   `(defadvice ,f (after emacspeak pre act comp)
