From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Wed, 9 Nov 2022 15:57:56 +0300
Subject: Kill and yank operations feedback enhancement

---
 lisp/emacspeak-advice.el | 48 ++++++++++++++++++++++++++++++------------------
 1 file changed, 30 insertions(+), 18 deletions(-)

diff --git a/lisp/emacspeak-advice.el b/lisp/emacspeak-advice.el
index cddd4d9..d2ed010 100644
--- a/lisp/emacspeak-advice.el
+++ b/lisp/emacspeak-advice.el
@@ -547,16 +547,31 @@ the words that were capitalized."
         (dtk-tone-deletion)
         (emacspeak-speak-region (point) start)))))
 
+(cl-loop for f in
+      '(delete-rectangle kill-rectangle)
+      do
+      (eval
+       `(defadvice ,f (after emacspeak pre act comp)
+          "Provide auditory feedback."
+          (when (ems-interactive-p)
+            (emacspeak-auditory-icon 'delete-object)))))
+
 ;;; Large deletions also produce auditory icons if possible
 
-(defadvice kill-line (before emacspeak pre act comp)
-  "Speak line before killing it. "
-  (when (ems-interactive-p)
-    (emacspeak-auditory-icon 'delete-object)
-    (when dtk-stop-immediately (dtk-stop))
-    (let ((dtk-stop-immediately nil))
-      (dtk-tone-deletion)
-      (emacspeak-speak-line 1))))
+(cl-loop for f in
+      '(kill-line
+        allout-kill-line
+        kill-sentence)
+      do
+      (eval
+       `(defadvice ,f (before emacspeak pre act comp)
+          "Speak line before killing it. "
+          (when (ems-interactive-p)
+            (when dtk-stop-immediately (dtk-stop))
+            (emacspeak-auditory-icon 'delete-object)
+            (let ((dtk-stop-immediately nil))
+              (dtk-tone-deletion)
+              (emacspeak-speak-line 1))))))
 
 (defadvice kill-sexp (before emacspeak pre act comp)
   "Speak the sexp you killed."
@@ -567,15 +582,6 @@ the words that were capitalized."
       (dtk-tone-deletion)
       (emacspeak-speak-sexp 1))))
 
-(defadvice kill-sentence (before emacspeak pre act comp)
-  "Speak the line you killed."
-  (when (ems-interactive-p)
-    (emacspeak-auditory-icon 'delete-object)
-    (when dtk-stop-immediately (dtk-stop))
-    (let ((dtk-stop-immediately nil))
-      (dtk-tone-deletion)
-      (emacspeak-speak-line 1))))
-
 (defadvice delete-blank-lines (before emacspeak pre act comp)
   "Provide auditory feedback."
   (when (ems-interactive-p)
@@ -2309,7 +2315,8 @@ Produce an auditory icon if possible."
 
 (cl-loop
  for f in
- '(yank yank-pop)
+ '(yank yank-pop
+        allout-yank allout-yank-pop)
  do
  (eval
   `(defadvice ,f (after emacspeak pre act comp)
@@ -2319,6 +2326,11 @@ Produce an auditory icon if possible."
        (emacspeak-auditory-icon 'yank-object)
        (emacspeak-speak-region (mark 'force) (point))))))
 
+(defadvice yank-rectangle (after emacspeak pre act comp)
+  "Produce an auditory icon if possible."
+  (when (ems-interactive-p)
+    (emacspeak-auditory-icon 'yank-object)))
+
 ;;}}}
 ;;{{{ advice non-incremental searchers
 
