From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Thu, 10 Nov 2022 07:38:59 +0300
Subject: Font lock toggling feedback

---
 lisp/emacspeak-advice.el | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/lisp/emacspeak-advice.el b/lisp/emacspeak-advice.el
index c065f0a..c7f91ab 100644
--- a/lisp/emacspeak-advice.el
+++ b/lisp/emacspeak-advice.el
@@ -2977,6 +2977,19 @@ Produce auditory icons if possible."
     (emacspeak-auditory-icon 'task-done)))
 
 ;;}}}
+;;{{{ Font lock mode toggling commands
+
+(defadvice font-lock-mode (after  emacspeak pre act comp)
+  "Provide an auditory icon if possible."
+  (when (ems-interactive-p)
+    (emacspeak-auditory-icon (if font-lock-mode 'on 'off))))
+
+(defadvice global-font-lock-mode (after emacspeak pre act comp)
+  "Provide an auditory icon if possible."
+  (when (ems-interactive-p)
+    (emacspeak-auditory-icon (if global-font-lock-mode 'on 'off))))
+
+;;}}}
 ;;{{{ Asking Questions:
 
 (defadvice yes-or-no-p (before emacspeak pre act comp)
