From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Thu, 10 Nov 2022 10:50:55 +0300
Subject: Removed duplicate

---
 lisp/emacspeak.el | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lisp/emacspeak.el b/lisp/emacspeak.el
index 60d9f43..2e5af53 100644
--- a/lisp/emacspeak.el
+++ b/lisp/emacspeak.el
@@ -268,7 +268,6 @@ that implements the speech-enabling extensions for `package' (a string)."
 (emacspeak-do-package-setup "paradox"'emacspeak-paradox)
 (emacspeak-do-package-setup "pcvs" 'emacspeak-pcl-cvs)
 (emacspeak-do-package-setup "elpy" 'emacspeak-elpy)
-(emacspeak-do-package-setup "elpy" 'emacspeak-elpy)
 (emacspeak-do-package-setup "pydoc" 'emacspeak-pydoc)
 (emacspeak-do-package-setup "python" 'emacspeak-python)
 (emacspeak-do-package-setup "python-mode" 'emacspeak-py)
