From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Tue, 22 Nov 2022 06:15:29 +0300
Subject: Auditory feedback on find completion

---
 lisp/emacspeak-find-dired.el | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lisp/emacspeak-find-dired.el b/lisp/emacspeak-find-dired.el
index f172e07..865b117 100644
--- a/lisp/emacspeak-find-dired.el
+++ b/lisp/emacspeak-find-dired.el
@@ -51,6 +51,10 @@
 
 ;;; Code:
 
+(defadvice  find-dired-sentinel (after emacspeak pre act )
+  "Provide auditory feedback"
+  (emacspeak-auditory-icon 'task-done))
+
 ;;}}}
 (provide 'emacspeak-find-dired)
 ;;{{{ end of file
