From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Tue, 22 Nov 2022 06:09:10 +0300
Subject: Proper voice assignment

---
 lisp/emacspeak-erc.el | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lisp/emacspeak-erc.el b/lisp/emacspeak-erc.el
index 7698ad3..96e762f 100644
--- a/lisp/emacspeak-erc.el
+++ b/lisp/emacspeak-erc.el
@@ -81,7 +81,7 @@ server."
    (erc-inverse-face voice-lighten-extra)
    (erc-underline-face voice-brighten-medium)
    (erc-prompt-face voice-bolden)
-   (erc-notice-face (quote inaudible))
+   (erc-notice-face voice-lighten)
    (erc-action-face voice-monotone)
    (erc-error-face voice-bolden-and-animate)
    (erc-dangerous-host-face voice-brighten-extra)
