From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Tue, 22 Nov 2022 06:41:50 +0300
Subject: Use forward-line for vertical movements

---
 lisp/emacspeak-ses.el | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lisp/emacspeak-ses.el b/lisp/emacspeak-ses.el
index ee16063..ca84332 100644
--- a/lisp/emacspeak-ses.el
+++ b/lisp/emacspeak-ses.el
@@ -124,13 +124,13 @@
 (defun emacspeak-ses-forward-row-and-summarize ()
   "Move to next row and summarize."
   (interactive)
-  (next-line)
+  (forward-line)
   (emacspeak-ses-summarize-current-cell))
 
 (defun emacspeak-ses-backward-row-and-summarize ()
   "Move to previous row  and summarize."
   (interactive)
-  (previous-line)
+  (forward-line -1)
   (emacspeak-ses-summarize-current-cell))
 
 ;;}}}
