From: "Igor B. Poretsky" <poretsky@rakurs.com>
Date: Wed, 23 Nov 2022 12:43:19 +0300
Subject: Fixed auditory icon duplication when reading lines

---
 lisp/emacspeak-speak.el | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lisp/emacspeak-speak.el b/lisp/emacspeak-speak.el
index cdf0e9f..610845c 100644
--- a/lisp/emacspeak-speak.el
+++ b/lisp/emacspeak-speak.el
@@ -967,7 +967,6 @@ with auditory icon `more'.  These can then be spoken using command
         (inhibit-point-motion-hooks t)
         (inhibit-modification-hooks t)
         (deactivate-mark nil)
-        (icon (get-char-property (point) 'auditory-icon))
         (before (get-char-property (point) 'before-string))
         (after (get-char-property (point) 'after-string))
         (display (get-char-property (point) 'display))
@@ -988,7 +987,6 @@ with auditory icon `more'.  These can then be spoken using command
      ((null arg))
      ((> arg 0) (setq start orig))
      (t (setq end orig)))
-    (when icon (emacspeak-auditory-icon icon))
     (when emacspeak-show-point
       (emacspeak-auditory-icon
        (cond
