From: "Igor B. Poretsky" <poretsky@rakurs.com>
Date: Wed, 23 Nov 2022 15:13:03 +0300
Subject: Optional audio feedback on minibuffer exit

---
 lisp/emacspeak-advice.el | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/lisp/emacspeak-advice.el b/lisp/emacspeak-advice.el
index 358c28b..7993eda 100644
--- a/lisp/emacspeak-advice.el
+++ b/lisp/emacspeak-advice.el
@@ -2656,6 +2656,12 @@ Produce auditory icons if possible."
 ;;}}}
 ;;{{{ setup minibuffer hooks:
 
+
+(defcustom emacspeak-confirm-minibuffer-exit nil
+  "Indicate minibuffer exit by auditory icon."
+  :type 'boolean
+  :group 'emacspeak)
+
 (defun emacspeak-minibuffer-setup-hook ()
   "Actions to take when entering the minibuffer with emacspeak running."
   (cl-declare (special minibuffer-exit-hook minibuffer-default))
@@ -2678,7 +2684,8 @@ Produce auditory icons if possible."
 (defun emacspeak-minibuffer-exit-hook ()
   "Actions performed when exiting the minibuffer with Emacspeak loaded."
   (dtk-stop)
-  (emacspeak-auditory-icon 'close-object))
+  (when emacspeak-confirm-minibuffer-exit
+    (emacspeak-auditory-icon 'close-object)))
 
 (add-hook 'minibuffer-exit-hook #'emacspeak-minibuffer-exit-hook)
 ;;}}}
