From: "Igor B. Poretsky" <poretsky@rakurs.com>
Date: Wed, 23 Nov 2022 15:11:31 +0300
Subject: Speak newline indentation

---
 lisp/emacspeak-advice.el | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/lisp/emacspeak-advice.el b/lisp/emacspeak-advice.el
index 819b3bb..685cba2 100644
--- a/lisp/emacspeak-advice.el
+++ b/lisp/emacspeak-advice.el
@@ -2012,11 +2012,15 @@ the newly created  line."
      (cond
       ((ems-interactive-p)
        (cond
-        (emacspeak-line-echo (emacspeak-speak-line))
-        (t
-         ad-do-it
-         (dtk-tone 225 75 'force)
-         (emacspeak-speak-line))))
+        (emacspeak-line-echo
+         (emacspeak-speak-line)
+         ad-do-it)
+        (t ad-do-it
+           (dtk-speak-using-voice voice-annotate
+                                  (format
+                                   "indent %s"
+                                   (current-column)))
+           (dtk-force))))
       (t ad-do-it))
      ad-return-value)))
 
