From: "Igor B. Poretsky" <poretsky@rakurs.com>
Date: Wed, 23 Nov 2022 15:16:17 +0300
Subject: More informative password prompt feedback

---
 lisp/emacspeak-advice.el | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/lisp/emacspeak-advice.el b/lisp/emacspeak-advice.el
index 0ca59f8..0877aef 100644
--- a/lisp/emacspeak-advice.el
+++ b/lisp/emacspeak-advice.el
@@ -896,9 +896,12 @@ icon."
 
 ;;}}}
 ;;{{{ advice various input functions to speak:
+
 (defadvice read-passwd (before emacspeak pre act comp)
-  "Provide auditory feedback."
-  (emacspeak-prompt "pwd"))
+  "Speak the prompt."
+  (dtk-stop)
+  (emacspeak-auditory-icon 'open-object)
+  (dtk-speak (ad-get-arg 0)))
 
 (defvar emacspeak-read-char-prompt-cache nil
   "Cache prompt from read-char and friends here for later introspection.")
