From: "Igor B. Poretsky" <poretsky@rakurs.com>
Date: Wed, 23 Nov 2022 17:18:27 +0300
Subject: Bbdb support enhancement

---
 lisp/emacspeak-bbdb.el | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/lisp/emacspeak-bbdb.el b/lisp/emacspeak-bbdb.el
index 702f7b8..3a41f93 100644
--- a/lisp/emacspeak-bbdb.el
+++ b/lisp/emacspeak-bbdb.el
@@ -129,12 +129,22 @@
     (emacspeak-auditory-icon 'close-object)
     (emacspeak-speak-line)))
 
+(defadvice bbdb (after emacspeak pre act)
+  "Provide auditory feedback"
+  (when (ems-interactive-p)
+    (emacspeak-auditory-icon 'search-hit)))
+
 (defadvice bbdb-bury-buffer (after emacspeak pre act)
   "Provide auditory feedback"
   (when (ems-interactive-p)
     (emacspeak-auditory-icon 'close-object)
     (emacspeak-speak-mode-line)))
 
+(defadvice bbdb-save-db (after emacspeak pre act)
+  "Produce an auditory icon if possible"
+  (when (ems-interactive-p)
+    (emacspeak-auditory-icon 'save-object)))
+
 (defadvice bbdb-elide-record (after emacspeak pre act)
   "Provide auditory feedback"
   (when (ems-interactive-p)
@@ -187,6 +197,11 @@
   (when (ems-interactive-p)
     (emacspeak-speak-other-window 1)))
 
+(defadvice bbdb/gnus-show-sender (after emacspeak pre act comp)
+  "Provide spoken feedback"
+  (when (ems-interactive-p)
+    (emacspeak-speak-other-window 1)))
+
 ;;}}}
 ;;{{{ silence messages 
 
