From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Thu, 24 Nov 2022 06:21:38 +0300
Subject: Personalities for old face names

---
 lisp/emacspeak-custom.el | 25 ++++++++++++++++++++++++-
 1 file changed, 24 insertions(+), 1 deletion(-)

diff --git a/lisp/emacspeak-custom.el b/lisp/emacspeak-custom.el
index a952eb3..a8c05dc 100644
--- a/lisp/emacspeak-custom.el
+++ b/lisp/emacspeak-custom.el
@@ -206,7 +206,30 @@
    (custom-set voice-smoothen-medium)
    (custom-state voice-smoothen)
    (custom-variable-button voice-animate)
-   (custom-variable-tag voice-bolden-medium)))
+   (custom-variable-tag voice-bolden-medium)
+   ;; Old names:
+   (custom-button-face voice-bolden)
+   (custom-button-pressed-face voice-bolden-extra)
+   (custom-button-pressed-unraised-face voice-bolden-extra)
+   (custom-button-mouse-face voice-bolden-medium)
+   (custom-button-unraised-face voice-smoothen)
+   (custom-changed-face voice-smoothen)
+   (custom-comment-face voice-monotone-medium)
+   (custom-comment-tag-face voice-monotone)
+   (custom-documentation-face voice-brighten-medium)
+   (custom-face-tag-face voice-lighten)
+   (custom-group-tag-face-1 voice-lighten-medium)
+   (custom-group-tag-face voice-bolden)
+   (custom-invalid-face voice-animate-extra)
+   (custom-link-face voice-bolden)
+   (custom-modified-face voice-lighten-medium)
+   (custom-rogue-face voice-bolden-and-animate)
+   (custom-modified-face voice-lighten-medium)
+   (custom-saved-face voice-smoothen-extra)
+   (custom-set-face voice-smoothen-medium)
+   (custom-state-face voice-smoothen)
+   (custom-variable-button-face voice-animate)
+   (custom-variable-tag-face voice-bolden-medium)))
 
 ;;}}}
 ;;{{{  custom navigation
