From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Thu, 24 Nov 2022 06:33:53 +0300
Subject: Dictionary support enhancement

---
 lisp/emacspeak-dictionary.el | 79 +++++++++++++++++++++++++++++++++++++++-----
 1 file changed, 71 insertions(+), 8 deletions(-)

diff --git a/lisp/emacspeak-dictionary.el b/lisp/emacspeak-dictionary.el
index 35f618f..9d25354 100644
--- a/lisp/emacspeak-dictionary.el
+++ b/lisp/emacspeak-dictionary.el
@@ -49,7 +49,23 @@
 ;;; server at dict.org:2628
 ;;; Code:
 ;;}}}
+;;{{{ Helper functions.
+
+(defun referenced-p ()
+  "Return t if the call was caused by selecting a link."
+  (cl-declare (special referenced-p))
+  (when (and (boundp 'referenced-p) referenced-p)
+    (setq referenced-p nil)
+    t))
+
+;;}}}
 ;;{{{ Advice interactive commands to speak.
+(defadvice link-selected (around emacspeak pre act comp)
+  "Get referenced calls to be recognizable."
+  (let ((referenced-p t))
+    ad-do-it)
+  ad-return-value)
+
 (defadvice dictionary (after emacspeak pre act comp)
   "Provide auditory feedback."
   (when (ems-interactive-p)
@@ -57,23 +73,60 @@
     (emacspeak-speak-mode-line)))
 (defadvice dictionary-close (after emacspeak pre act comp)
   "Provide auditory feedback."
-  (when (ems-interactive-p)
+  (when (or (ems-interactive-p) (referenced-p))
+    (emacspeak-auditory-icon 'close-object)
+    (when (and (boundp 'dictionary-window-configuration)
+               (boundp 'dictionary-selected-window))
+      (let ((window-configuration dictionary-window-configuration)
+            (selected-window dictionary-selected-window))
+        (replace-buffer-in-windows)
+        (when window-configuration
+          (set-window-configuration window-configuration))
+        (when selected-window
+          (select-window selected-window))))
+    (emacspeak-speak-mode-line)))
+(defadvice dictionary-restore-state (after emacspeak pre act comp)
+  "Provide auditory feedback."
+  (when (referenced-p)
     (emacspeak-auditory-icon 'close-object)
     (emacspeak-speak-mode-line)))
 (defadvice dictionary-select-dictionary (after emacspeak pre act comp)
   "Provide auditory feedback."
-  (when (ems-interactive-p)
-    (emacspeak-auditory-icon 'select-object)
-    (message "Selected dictionary")))
+  (when (or (ems-interactive-p) (referenced-p))
+    (emacspeak-auditory-icon 'open-object)
+    (emacspeak-speak-line)))
 (defadvice dictionary-select-strategy (after emacspeak pre act comp)
   "Provide auditory feedback."
-  (when (ems-interactive-p)
+  (when (or (ems-interactive-p) (referenced-p))
+    (emacspeak-auditory-icon 'open-object)
+    (emacspeak-speak-line)))
+
+(defadvice dictionary-set-dictionary (around emacspeak pre act comp)
+  "Provide auditory feedback."
+  (if (not (referenced-p))
+      ad-do-it
     (emacspeak-auditory-icon 'select-object)
-    (message "Selected strategy")))
+    (let ((emacspeak-speak-messages t))
+      ad-do-it))
+  ad-return-value)
+
+(defadvice dictionary-set-strategy (around emacspeak pre act comp)
+  "Provide auditory feedback."
+  (if (not (referenced-p))
+      ad-do-it
+    (emacspeak-auditory-icon 'select-object)
+    (let ((emacspeak-speak-messages t))
+      ad-do-it))
+  ad-return-value)
 
 (defadvice dictionary-search (after emacspeak pre act comp)
   "Provide auditory feedback."
-  (when (ems-interactive-p)
+  (when (or (ems-interactive-p) (referenced-p))
+    (emacspeak-auditory-icon 'search-hit)
+    (emacspeak-speak-line)))
+(defadvice dictionary-new-search (after emacspeak pre act comp)
+  "Provide auditory feedback."
+  (when (or (ems-interactive-p) (referenced-p))
     (emacspeak-auditory-icon 'search-hit)
     (emacspeak-speak-line)))
 (defadvice dictionary-lookup-definition (after emacspeak pre act comp)
@@ -84,7 +137,7 @@
 
 (defadvice dictionary-match-words (after emacspeak pre act comp)
   "Provide auditory feedback."
-  (when (ems-interactive-p)
+  (when (or (ems-interactive-p) (referenced-p))
     (emacspeak-auditory-icon 'search-hit)
     (emacspeak-speak-line)))
 
@@ -106,6 +159,16 @@
     (emacspeak-speak-text-range 'link-function)))
 
 ;;}}}
+;;{{{ mapping font faces to personalities 
+
+(voice-setup-add-map
+ '(
+   (dictionary-button-face voice-bolden)
+   (dictionary-word-entry-face voice-animate)
+   (dictionary-reference-face voice-bolden)
+   ))
+
+;;}}}
 (provide 'emacspeak-dictionary)
 ;;{{{ end of file
 
