From: "Igor B. Poretsky" <poretsky@rakurs.com>
Date: Thu, 24 Nov 2022 16:13:48 +0300
Subject: Org mode support enhancement

---
 lisp/emacspeak-org.el | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lisp/emacspeak-org.el b/lisp/emacspeak-org.el
index 36e4781..4229615 100644
--- a/lisp/emacspeak-org.el
+++ b/lisp/emacspeak-org.el
@@ -215,6 +215,7 @@
  '(
    org-delete-indentation
    org-insert-heading org-insert-todo-heading
+   org-insert-subheading org-insert-todo-subheading
    org-promote-subtree org-demote-subtree
    org-do-promote org-do-demote
    org-move-subtree-up org-move-subtree-down
@@ -469,6 +470,7 @@
  for f in
  '(
    org-occur
+   org-back-to-heading
    org-beginning-of-line org-end-of-line
    org-beginning-of-item org-beginning-of-item-list)
  do
