From: "Igor B. Poretsky" <poretsky@rakurs.com>
Date: Fri, 25 Nov 2022 09:55:18 +0300
Subject: Emacs customization support fixes

---
 lisp/emacspeak-custom.el | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/lisp/emacspeak-custom.el b/lisp/emacspeak-custom.el
index a8c05dc..e0d07a9 100644
--- a/lisp/emacspeak-custom.el
+++ b/lisp/emacspeak-custom.el
@@ -235,7 +235,7 @@
 ;;{{{  custom navigation
 
 (defvar emacspeak-custom-group-regexp
-  "^/-"
+  "^[^:]+ group:"
   "Pattern identifying start of custom group.")
 
 ;;;###autoload
@@ -243,15 +243,17 @@
   "Jump to custom group when in a customization buffer."
   (interactive)
   (cl-declare (special emacspeak-custom-group-regexp))
-  (when (eq major-mode 'custom-mode)
+  (when (eq major-mode 'Custom-mode)
     (goto-char (point-min))
     (re-search-forward emacspeak-custom-group-regexp
                        nil t)
-    (emacspeak-auditory-icon 'large-movement)
-    (emacspeak-speak-line)))
+    (forward-line 0)
+    (when (called-interactively-p 'interactive)
+      (emacspeak-auditory-icon 'large-movement)
+      (emacspeak-speak-line))))
 
 (defvar emacspeak-custom-toolbar-regexp
-  "^Operate on everything in this buffer:"
+  "^ *Operate on \\(everything\\|all settings\\) in this buffer"
   "Pattern that identifies toolbar section.")
 
 
@@ -259,12 +261,14 @@
   "Jump to custom toolbar when in a customization buffer."
   (interactive)
   (cl-declare (special emacspeak-custom-toolbar-regexp))
-  (when (eq major-mode 'custom-mode)
+  (when (eq major-mode 'Custom-mode)
     (goto-char (point-min))
     (re-search-forward emacspeak-custom-toolbar-regexp nil
                        t)
-    (emacspeak-auditory-icon 'large-movement)
-    (emacspeak-speak-line)))
+    (forward-line 0)
+    (when (called-interactively-p 'interactive)
+      (emacspeak-auditory-icon 'large-movement)
+      (emacspeak-speak-line))))
 
 ;;}}}
 ;;{{{  bind emacspeak commands 
