From: "Igor B. Poretsky" <poretsky@rakurs.com>
Date: Fri, 25 Nov 2022 15:47:23 +0300
Subject: Removed redundant function

---
 lisp/emacspeak-speak.el | 24 ------------------------
 1 file changed, 24 deletions(-)

diff --git a/lisp/emacspeak-speak.el b/lisp/emacspeak-speak.el
index 86719c2..faed44e 100644
--- a/lisp/emacspeak-speak.el
+++ b/lisp/emacspeak-speak.el
@@ -3288,30 +3288,6 @@ Argument O specifies overlay."
                (1+ (- orig start)) (- end start))))))
 
 ;;}}}
-;;{{{  translate faces to voices
-
-(defun voice-lock-voiceify-faces ()
-  "Map faces to personalities."
-  (save-excursion
-    (goto-char (point-min))
-    (let ((inhibit-read-only t)
-          (face nil)
-          (start (point)))
-      (unwind-protect
-          (while (not (eobp))
-            (setq face (get-text-property (point) 'face))
-            (goto-char
-             (or (next-single-property-change (point) 'face)
-                 (point-max)))
-            (put-text-property start (point)
-                               'personality
-                               (if (listp face)
-                                   (car face)
-                                 face))
-            (setq start (point)))
-        (setq inhibit-read-only nil)))))
-
-;;}}}
 ;;{{{  completion helpers
 
 ;;{{{ switching to completions window from minibuffer:
