From: "Igor B. Poretsky" <poretsky@rakurs.com>
Date: Fri, 25 Nov 2022 16:02:09 +0300
Subject: Omit text properties

---
 lisp/emacspeak-speak.el | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lisp/emacspeak-speak.el b/lisp/emacspeak-speak.el
index bc84f2f..655e969 100644
--- a/lisp/emacspeak-speak.el
+++ b/lisp/emacspeak-speak.el
@@ -1589,7 +1589,7 @@ Negative prefix arg speaks from start of buffer to point."
                (previous-single-property-change beg 'mouse-face)
                (point-min)))
     (setq end (or (next-single-property-change end 'mouse-face) (point-max)))
-    (buffer-substring beg end)))
+    (buffer-substring-no-properties beg end)))
 
 ;;}}}
 ;;{{{ mail check
