From: "Igor B. Poretsky" <poretsky@rakurs.com>
Date: Fri, 25 Nov 2022 16:14:34 +0300
Subject: Removed redundant directive

---
 lisp/emacspeak-speak.el | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lisp/emacspeak-speak.el b/lisp/emacspeak-speak.el
index 204dc8c..0d722b4 100644
--- a/lisp/emacspeak-speak.el
+++ b/lisp/emacspeak-speak.el
@@ -1560,8 +1560,6 @@ Negative prefix arg speaks from start of buffer to point."
         (dtk-speak "First ask for help")))))
 
 ;;;###autoload
-
-;;;###autoload
 (defun emacspeak-speak-minibuffer (&optional arg)
   "Speak the minibuffer contents
  With prefix arg, speaks the rest of the buffer from point.
