From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Sat, 26 Nov 2022 07:53:34 +0300
Subject: Timezone entering fix

---
 lisp/emacspeak-speak.el | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lisp/emacspeak-speak.el b/lisp/emacspeak-speak.el
index d0e6ed9..51cb98c 100644
--- a/lisp/emacspeak-speak.el
+++ b/lisp/emacspeak-speak.el
@@ -2124,6 +2124,7 @@ Optional second arg `set' sets the TZ environment variable as well."
   (interactive
    (list
     (let ((completion-ignore-case t)
+          (insert-default-directory nil)
           (ido-case-fold t)
           (read-file-name-completion-ignore-case t))
       (read-file-name
