From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Sat, 26 Nov 2022 08:04:10 +0300
Subject: Minibuffer completion keymap fix

---
 lisp/emacspeak-speak.el | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lisp/emacspeak-speak.el b/lisp/emacspeak-speak.el
index 83b8b5f..3389f6d 100644
--- a/lisp/emacspeak-speak.el
+++ b/lisp/emacspeak-speak.el
@@ -3687,7 +3687,7 @@ This command  is designed for use in a windowing environment like X."
 (define-key minibuffer-local-completion-map "\C-p" 'emacspeak-minibuffer-previous-completion)
 (define-key minibuffer-local-completion-map (kbd "C-@") 'emacspeak-minibuffer-choose-completion)
 (define-key minibuffer-local-completion-map
-  (kbd "C-SPC") 'emacspeak-minibuffer-choose-completion)
+  (kbd "<C-return>") 'emacspeak-minibuffer-choose-completion)
 
 ;;}}}
 ;;{{{ Open Emacspeak Info Pages:
