From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Sat, 26 Nov 2022 08:54:47 +0300
Subject: G-client load path fix

---
 lisp/g-client/g-load-path.el | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/lisp/g-client/g-load-path.el b/lisp/g-client/g-load-path.el
index 657ec93..88b6c7c 100644
--- a/lisp/g-client/g-load-path.el
+++ b/lisp/g-client/g-load-path.el
@@ -38,6 +38,17 @@
        (file-name-directory load-file-name))
   "Directory where g-client  is built. ")
 
+(defvar ems-directory
+  (and load-file-name
+       (file-name-as-directory
+        (expand-file-name ".."
+                          (file-name-directory load-file-name))))
+  "Directory where emacspeak is built. ")
+
+(unless (member ems-directory load-path)
+  (setq load-path
+        (cons ems-directory load-path)))
+
 (unless (member g-directory load-path)
   (setq load-path
         (cons g-directory load-path)))
