From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Tue, 3 Oct 2023 15:06:09 +0300
Subject: Eliminate query on exit

---
 lisp/dtk-speak.el | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lisp/dtk-speak.el b/lisp/dtk-speak.el
index c1d9450..793da0a 100644
--- a/lisp/dtk-speak.el
+++ b/lisp/dtk-speak.el
@@ -1720,6 +1720,7 @@ program. Port defaults to dtk-local-server-port"
     (set-process-coding-system process 'utf-8 'utf-8)
     (tts-env-set-process-env process
                              (tts-env-get (tts-env-key dtk-program)))
+    (set-process-query-on-exit-flag process nil)
     process))
 
 (defun dtk-initialize ()
