From: "Igor B. Poretsky" <poretsky@mlbox.ru>
Date: Sun, 8 Jan 2023 09:48:29 +0300
Subject: Don't require deprecated package

---
 lisp/emacspeak-smartparens.el | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lisp/emacspeak-smartparens.el b/lisp/emacspeak-smartparens.el
index 0517488..4b6b5c3 100644
--- a/lisp/emacspeak-smartparens.el
+++ b/lisp/emacspeak-smartparens.el
@@ -51,7 +51,6 @@
 
 ;;}}}
 ;;{{{  Required modules
-(eval-when-compile (require 'cl))
 (require 'cl-lib)
 (cl-declaim  (optimize  (safety 0) (speed 3)))
 (require 'emacspeak-preamble)
