diff --git a/lisp/emacspeak-tetris.el b/lisp/emacspeak-tetris.el
index 1b9fa82..52f74ab 100644
--- a/lisp/emacspeak-tetris.el
+++ b/lisp/emacspeak-tetris.el
@@ -48,13 +48,22 @@
 
 (cl-declaim  (optimize  (safety 0) (speed 3)))
 (require 'emacspeak-preamble)
-(require 'tetris "tetris" 'no-error)
+(require 'tetris () 'no-error)
 ;;}}}
 ;;{{{  Introduction 
 
 ;;; Auditory interface to tetris
 
 ;;}}}
+;;{{{ Forward declarations
+
+(declare-function tetris-erase-shape "tetris.el" ())
+(declare-function tetris-test-shape "tetris.el" ())
+(declare-function tetris-draw-shape "tetris.el" ())
+(declare-function tetris-shape-width "tetris.el" ())
+(declare-function gamegrid-get-cell "gamegrid.el" (x y))
+
+;;}}}
 ;;{{{  Setup speech display table 
 
 (defvar emacspeak-tetris-pronunciations-defined nil
@@ -330,7 +339,7 @@ compensate for the difference between aural and visual
 interaction.  Despite the clock being slowed down, playing
 Tetris with speech feedback alone requires a lot of
 concentration and the game is a good mental challenge. "
-  (cl-declare (special tetris-tick-period
+  (cl-declare (special tetris-tick-period tetris-height
                        emacspeak-tetris-blank-row))
   (when (ems-interactive-p)
     (setq tetris-tick-period emacspeak-tetris-tick-period)
