diff -au emacspeak-tetris.el.orig emacspeak-tetris.el
--- emacspeak-tetris.el.orig	2018-03-20 16:29:07.913143027 +0300
+++ emacspeak-tetris.el	2018-03-20 16:44:55.905755330 +0300
@@ -1,4 +1,4 @@
-;;; emacspeak-tetris.el --- Speech enable game of Tetris  -*- lexical-binding: t; -*-
+;;; emacspeak-tetris.el --- Speech enable game of Tetris
 ;;; $Id$
 ;;; $Author: tv.raman.tv $ 
 ;;; Description: Auditory interface to tetris
@@ -54,6 +54,15 @@
 ;;; Auditory interface to tetris
 
 ;;}}}
+;;{{{ Forward declarations
+
+(declare-function tetris-erase-shape "tetris.el" ())
+(declare-function tetris-test-shape "tetris.el" ())
+(declare-function tetris-draw-shape "tetris.el" ())
+(declare-function tetris-shape-width "tetris.el" ())
+(declare-function gamegrid-get-cell "gamegrid.el" (x y))
+
+;;}}}
 ;;{{{  Setup speech display table 
 
 (defvar emacspeak-tetris-pronunciations-defined nil
@@ -328,7 +337,7 @@
 interaction.  Despite the clock being slowed down, playing
 Tetris with speech feedback alone requires a lot of
 concentration and the game is a good mental challenge. "
-  (declare (special tetris-tick-period
+  (declare (special tetris-tick-period tetris-height
                     emacspeak-tetris-blank-row))
   (when (ems-interactive-p)
     (setq tetris-tick-period emacspeak-tetris-tick-period)
